iocore/net/UnixNetVConnection.cc

```diff
@@ -270,6 +270,21 @@ read_from_net(NetHandler *nh, UnixNetVConnection *vc, EThread *thread)
     close_UnixNetVConnection(vc, thread);
     return;
   }
+
+  if (vc->read.error) {
+    int err = 0, errlen = sizeof(int);
+    if (getsockopt(vc->con.fd, SOL_SOCKET, SO_ERROR, &err, (socklen_t *)&errlen) == -1) {
+      err = errno;
+    }
+    // if it is a non-temporary error, we should die appropriately
+    if (err != EAGAIN && err != EINTR) {
+      read_signal_error(nh, vc, err);
+      return;
+    } else {
+      vc->read.error = 0;
```

Comment:
@jpeach The socket fd is not valid descriptor if getsockopt failed. It is fatal error and we should signal_error to SM and close the NetVC.